packages/onchain/contracts/TorchShip.sol

```diff
@@ -100,19 +111,55 @@ contract TorchShip is Initializable, ContextUpgradeSafe, OwnableUpgradeSafe {
     }
   }
 
+  // update the totalSupply for the observation at the current timestamp. each observation is updated at most
+  // once per epoch period.
```

Comment:
it is the window size divided by the granularity. as in this formula: `block.number.sub(lastBlockHeight) > windowSize.div(granularity)`.

so in a window size of 7 days, and granularity of 1 day, this would be 24 hours.